include/zephyr/drivers/misc/pio_rpi_pico/pio_rpi_pico.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023 Tokita, Hiroshi <tokita.hiroshi@fujitsu.com>
+ * Copyright (c) 2023 Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_MISC_PIO_PICO_RPI_PIO_PICO_RPI_H_
+#define ZEPHYR_DRIVERS_MISC_PIO_PICO_RPI_PIO_PICO_RPI_H_
+
+#include <zephyr/devicetree/gpio.h>
+
+#include <hardware/pio.h>
+
+/**
+ * @brief Utility macro to declare a PIO program. The program is a list
+ *        of 16 bit instructions, generated by the pioasm tool.
+ *
+ * @param name Name of the program.
+ * @param wrap_target Wrap target as specified by the PIO program.
+ * @param wrap Wrap source as specified by the PIO program.
+ * @param ... Comma separated list of PIO instructions.
+ */
+#define RPI_PICO_PIO_DECLARE_PROGRAM(name, wrap_target, wrap, ...)	\
+	static const uint32_t name ## _wrap_target = wrap_target;	\
+	static const uint32_t name ## _wrap = wrap;			\
+	static const uint16_t name ## _program_instructions[] = {	\
+		__VA_ARGS__						\
+	};								\
+	static const struct pio_program name ## _program = {		\
```

Comment:
Should this be called just `name`? (as in, drop the `_program` suffix) That way it's a bit easier to understand that this is what it taken by `pio_add_program` as an argument just by reading the code.

Reply:
I don't know what it will be useful for, we can always add it later

Reply:
Oh sorry, I though you meant a getter for the instructions themselves. I'll add it

Reply:
done